File: problems/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> s; // <num, idx>

        int i = 0;
        for (int& n : nums) {

            int diff = target - n;
            if (s.find(diff) != s.end()) {
                return {s[diff], i};
            }

            s.insert(std::make_pair(n, i));
            i++;
        }

        return {-1, -1};
    }
};

File: problems/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        idxMap = {} # key: n, val: idx
        for i, n in enumerate(nums):
            diff = target - n

            if diff in idxMap:
                return [idxMap[diff], i]
            else:
                idxMap[n] = i
